docs/concepts/webhooks.md

```diff
@@ -0,0 +1,36 @@
+# Webhooks
+
+The [etcd-druid controller-manager](controllers.md#controller-manager) registers certain [admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) that allow for validation or mutation of requests on resources in the cluster, in order to prevent misconfiguration and restrict access to the etcd cluster resources.
+
+All webhooks that are a part of etcd-druid reside in package `internal/webhook`, as sub-packages.
+
+## Package Structure
+
+The typical package structure for the webhooks that are part of etcd-druid is shown with the *sentinel webhook*:
+
+``` bash
+internal/webhook/sentinel
+├── config.go
+├── handler.go
+└── register.go
+```
+
+- `config.go`: contains all the logic for the configuration of the webhook, including feature gate activations, CLI flag parsing and validations.
+- `register.go`: contains the logic for registering the webhook with the etcd-druid controller manager.
+- `handler.go`: contains the webhook admission handler logic.
+
+Each webhook package may also contain auxiliary files which are relevant to that specific webhook.
+
+## Sentinel Webhook
+
+Druid controller-manager registers and runs the [etcd controller](controllers.md#etcd-controller), which creates and manages various resources such as `Leases`, `ConfigMap`s, and the `Statefulset` for the etcd cluster. It is essential for all these resources to contain correct configuration for the proper functioning of the etcd cluster.
+
+Unintended changes to any of these *managed resources* can lead to misconfiguration of the etcd cluster, leading to unwanted downtime for etcd traffic. To prevent such unintended changes, a validating webhook called *sentinel webhook* guards these managed resources, ensuring that only authorized entities can perform operations on these managed resources.
+
+*Sentinel webhook* prevents *UPDATE* and *DELETE* operations on all resources managed by *etcd controller*, unless such an operation is performed by druid itself, and during reconciliation of the `Etcd` resource. Operations are also allowed if performed by one of the authorized entities specified by CLI flag `--sentinel-exempt-service-accounts`.
```

Comment:
Does the sentinel webhook allow exempt service accounts to make changes during the reconciliation process

Reply:
No. During reconciliation, only druid will be allowed to modify the managed resources. This is to ensure that there are no other unintended operations in between reconciliation from other actors. But outside of reconciliation, exempt serviceaccounts can make changes, since it doesn't affect druid's reconciliation in any way.

I will also document this behavior, now that you mentioned it.

Reply:
If an exempt serviceaccount really wants to modify managed resources, they will need to have a retry logic, and attempt to retry their operation until after druid is finished with reconciliation.